### PR TITLE
Conductor: Implement NotNotarisedBlockExtension test case

### LINKED ISSUE #692: R&D: Conductor: Leader extends not notarized prev_block
Implement a test: Leader extends not notarized prev_block.
```
(T0) Leader_0: send Proposal0_0, extends not notarized prev_block
(T0) Leader_1: send Proposal1_0
(T0 + δ) Replica_0: reject Proposal0_, verify Proposal1_0
```

### PR DESCRIPTION
**Pull request by MurashovVen**
Implementing test case:

```
// Leader extends not notarized prev_block
(T0) Leader_0: send Proposal0_0, extends not notarized prev_block
(T0) Leader_1: send Proposal1_0
(T0 + δ) Replica_0: reject Proposal0_, verify Proposal1_0
```

### CODE REVIEW

File: code/go/0chain.net/conductor/cases/not_notarised_block_extension.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+func (n *NotNotarisedBlockExtension) check() (success bool, err error) {
+	if n.mockedBlockHashToExtend == "" {
+		return false, errors.New("mocked block is nil")
+	}
+
+	for hash, status := range n.result {
+		switch {
+		case hash == n.mockedBlockHashToExtend && status == block.VerificationSuccessful:
+			msg := fmt.Sprintf("block with %s previous block hash have unexpected status: %d", hash, status)
```
Comment (dabasov) [inline code/go/0chain.net/conductor/cases/not_notarised_block_extension.go]: typo, should be "previous block hash has"

Comment (dabasov) [reply code/go/0chain.net/conductor/cases/not_notarised_block_extension.go]: why not `return false, fmt.Errorf()..`

File: code/go/0chain.net/miner/protocol_block_main.go [hunk 2]
```diff
@@ -307,3 +307,8 @@ func (mc *Chain) GenerateBlock(ctx context.Context, b *block.Block,
 	node.Self.Underlying().Info.AvgBlockTxns = int(math.Round(bsHistogram.Mean()))
 	return nil
 }
+
+/*UpdateFinalizedBlock - update the latest finalized block */
```
Comment (dabasov) [inline code/go/0chain.net/miner/protocol_block_main.go:311]: why do we need this func?

File: code/go/0chain.net/conductor/config/execute.go [hunk 3]
```diff
@@ -78,6 +78,12 @@ type Executor interface {
 	StorageTree(st *Bad) (err error)
 	ValidatorProof(vp *Bad) (err error)
 	Challenges(cs *Bad) (err error)
+
+	// ConfigureNotNotarisedBlockExtensionCheck prepares state for cases.NotNotarisedBlockExtension test case.
+	ConfigureNotNotarisedBlockExtensionCheck(*ExtendNotNotarisedBlock) (err error)
```
Comment (dabasov) [inline code/go/0chain.net/conductor/config/execute.go:83]: this names looks a little bit wired comparing to others, why so?

File: code/go/0chain.net/miner/protocol_block_main.go (hunk 2 above)
Comment (MurashovVen) [reply code/go/0chain.net/miner/protocol_block_main.go:311]: The main implementation (`updateFinalizedBlock`) stays at the base code part shared to all builds. Wrapping functions like that help ignore useless code copying.

Comment (MurashovVen) [reply code/go/0chain.net/miner/protocol_block_main.go:311]: The main implementation (`updateFinalizedBlock`) stays at the base code part shared to all builds. Wrapping functions like that help ignore useless code copying.

File: code/go/0chain.net/conductor/cases/not_notarised_block_extension.go (hunk 1 above)
Comment (MurashovVen) [reply code/go/0chain.net/conductor/cases/not_notarised_block_extension.go]: Fixed.

**Review by dabasov — COMMENTED** [review]
lgtm

**Pull request merged**